guvnor-project/guvnor-project-backend/src/main/java/org/guvnor/common/services/project/backend/server/AbstractProjectService.java

```diff
@@ -157,198 +166,204 @@ public WorkingSetSettings loadWorkingSetConfig( final Path project ) {
+            final org.uberfire.java.nio.file.Path newProjectPath = projectDirectory.resolveSibling(newName);
 
-            final POM content = pomService.load( pathToPomXML );
+            final POM content = pomService.load(pathToPomXML);
 
-            if ( newProjectPath.equals( projectDirectory ) ) {
+            if (newProjectPath.equals(projectDirectory)) {
                 return pathToPomXML;
             }
 
-            if ( ioService.exists( newProjectPath ) ) {
-                throw new FileAlreadyExistsException( newProjectPath.toString() );
+            if (ioService.exists(newProjectPath)) {
+                throw new FileAlreadyExistsException(newProjectPath.toString());
             }
 
-            final Path oldProjectDir = Paths.convert( projectDirectory );
-            final Project oldProject = resourceResolver.resolveProject( oldProjectDir );
+            final Path oldProjectDir = Paths.convert(projectDirectory);
+            final Project oldProject = resourceResolver.resolveProject(oldProjectDir);
 
-            content.setName( newName );
-            final Path newPathToPomXML = Paths.convert( newProjectPath.resolve( POM_PATH ) );
+            content.setName(newName);
+            final Path newPathToPomXML = Paths.convert(newProjectPath.resolve(POM_PATH));
             try {
-                ioService.startBatch( newProjectPath.getFileSystem() );
-                ioService.move( projectDirectory,
-                                newProjectPath,
-                                commentedOptionFactory.makeCommentedOption( comment ) );
-                pomService.save( newPathToPomXML,
-                                 content,
-                                 null,
-                                 comment );
-            } catch ( final Exception e ) {
+                ioService.startBatch(newProjectPath.getFileSystem());
+                ioService.move(projectDirectory,
+                               newProjectPath,
+                               commentedOptionFactory.makeCommentedOption(comment));
+                pomService.save(newPathToPomXML,
+                                content,
+                                null,
+                                comment);
+            } catch (final Exception e) {
                 throw e;
             } finally {
+                final Project newProject = resourceResolver.resolveProject(Paths.convert(newProjectPath));
+                renameProjectEvent.fire(new RenameProjectEvent(oldProject,
+                                                               newProject));
                 ioService.endBatch();
             }
-            final Project newProject = resourceResolver.resolveProject( Paths.convert( newProjectPath ) );
-            invalidateDMOCache.fire( new InvalidateDMOProjectCacheEvent( sessionInfo, oldProject, oldProjectDir ) );
-            renameProjectEvent.fire( new RenameProjectEvent( oldProject, newProject ) );
```

Comment:
Real change is to move this line (and the `newProject` declaration) inside the `finally` block, so it is executed before the `ioService.endBatch();` call. This way, the `renameProjectEvent` is fired before the `deleteProjectEvent`, which is fired when the `pom.xml` is actually removed.